core/src/test/scala/integration/kafka/admin/TopicCommandIntegrationTest.scala

```diff
@@ -732,7 +733,18 @@ class TopicCommandIntegrationTest extends KafkaServerTestHarness with Logging wi
     try {
       killBroker(0)
       val aliveServers = brokers.filterNot(_.config.brokerId == 0)
-      TestUtils.waitForPartitionMetadata(aliveServers, underMinIsrTopic, 0)
+
+      if (isKRaftTest()) {
+        TestUtils.ensureConsistentKRaftMetadata(aliveServers, controllerServer, "Timeout waiting for topic configs propagating to brokers")
+      } else {
+        TestUtils.waitUntilTrue(
```

Comment:
Since this test only fails for KRaft mode and zk will be deprecated in 3.4.0 and totally removed in 4.0.0 I'm inclined to leave it as it was, but it's also OK to improve it.